### DISCUSSION TITLE
Highlight all Matches in Scroll Bar

### DISCUSSION THREAD

**Discussion by PaltryProgrammer** [dnGrep/dnGrep#634] [Ideas]
Thank You for providing dnGrep . I am quite impressed w/ its option to display matched files and the quickness of response in doing so and of the very convenient highlighting of matched text and providing the number of said matches via the file list . However when multiple matches occur one is left wondering where those beyond the first are located . May I suggest for your kind consideration several methods to wit i.e. e.g. Next/Previous buttons , highlights in vertical scroll bar . Thank You Kindly - Cheerio

**Comment by doug24** [comment]
Which scroll bar - the one in the results tree showing all files/matches, or the preview window showing the currently selected file?

Because the results tree has only matching and context lines, the markers would be pretty much the whole scroll bar.  I do see where scroll bar markers could be useful in the preview window.

Just in case you didn't know: the F3 and F4 keys step forward and backward through the matches, and as each match is selected, the file will load in the preview window, and scroll to the matching line.

**Reply by PaltryProgrammer** [reply]
I was referring to the vertical scroll bar in the Preview Window . Thank You for informing of the F3/F4 stepping . This is very useful to me . Upon utilizing it I observed the matched text and surrounding file is displayed in both the Results window and Preview window . I do not understand the need for such duplication . Further , doing so prevents me from knowing which file in the Results window is being displayed as the Results list is now out of view and as only the name of the file at the top of the Preview window is displayed w/o the full path . Further complicated my usage is the fact I am searching through a history of backups so each file name is identical . I expected F3/F4 to only highlight the file name in the Results window while displaying the highlighted matched line in the Preview Window . Thank You . PS If a "Buy Me a Cup of Coffee" button were available I would cheerfully do so as I find dnGrep useful and impressive . Kind Regards

**Comment by doug24** [comment]
New feature in v3.0.29.0
